src/SDKs/Batch/DataPlane/Azure.Batch/Protocol/BatchSharedKeyCredential.cs

```diff
@@ -74,24 +74,22 @@ public override Task ProcessHttpRequestAsync(HttpRequestMessage httpRequest, Can
             signature.Append(httpRequest.Content != null && httpRequest.Content.Headers.Contains("Content-Language") ? httpRequest.Content.Headers.GetValues("Content-Language").FirstOrDefault() : string.Empty).Append('\n');
 
             // Handle content length
-            if (httpRequest.Content != null)
+            long? contentLength = httpRequest.Content?.Headers?.ContentLength;
+            if (httpRequest.Method == HttpMethod.Delete && contentLength == null)
             {
-                signature.Append(httpRequest.Content.Headers.ContentLength.HasValue ? httpRequest.Content.Headers.ContentLength.ToString() : string.Empty).Append('\n');
+                // Because C# httpRequest adds a content-length = 0 header for DELETE even if there is no body (but only in netframework), we have to
+                // sign the request knowing that there will be content-length set.
+#if FullNetFx
+                contentLength = 0;
```

Comment:
I'd still prefer that the `#if FullNetFx` was outside of the if statement so we're not generating an empty if for dotNetCore.